test/bridge.test.ts

```diff
@@ -627,6 +637,84 @@ describe('validating inbound swaps', () => {
     );
   });
 
+  test('validates that supplier cannot frontrun', async () => {
+    const supplierInfo = (await t.rov(contract.getSupplier(0n)))!;
+
+    // Test base fee
+    await t.txOk(
+      contract.updateSupplierFees(
+        supplierInfo['inbound-fee'],
+        supplierInfo['outbound-fee'],
+        supplierInfo['outbound-base-fee'],
+        supplierInfo['inbound-base-fee'] + 10n
+      ),
+      supplier
+    );
+
+    const receipt = await t.txErr(
+      contract.escrowSwap(
+        { header: Buffer.from([]), height: 1n },
+        [],
+        txHex,
+        proof,
+        0n,
+        htlc.senderPublicKey,
+        htlc.recipientPublicKey,
+        CSV_DELAY_BUFF,
+        hash,
+        swapperHex,
+        0n,
+        xbtcAmount
+      ),
+      swapper
+    );
+
+    expect(receipt.value).toEqual(27n);
+
+    // Test inbound fee
+    const inFee = supplierInfo['inbound-fee'];
+    const newInFee = inFee === null ? inFee : inFee + 10n;
+    await t.txOk(
+      contract.updateSupplierFees(
+        newInFee,
```

Comment:
```suggestion
        supplierInfo['inbound-fee']! + 10n,
```

Just an FYI, with Typescript you can use the `!` operator here too. Not a big deal, just wanted to mention.